tests/integration/devfile/cmd_dev_test.go

```diff
@@ -240,17 +240,21 @@ var _ = Describe("odo dev command tests", func() {
 
 		When("odo dev is executed and Ephemeral is set to false", func() {
 
+			var devSession helper.DevSession
 			BeforeEach(func() {
 				helper.Cmd("odo", "preference", "set", "Ephemeral", "false").ShouldPass()
-
-				devSession, _, _, _, err := helper.StartDevMode()
+				var err error
+				devSession, _, _, _, err = helper.StartDevMode()
 				Expect(err).ToNot(HaveOccurred())
-				defer devSession.Kill()
 				// An ENV file should have been created indicating current namespace
 				Expect(helper.VerifyFileExists(".odo/env/env.yaml")).To(BeTrue())
 				helper.FileShouldContainSubstring(".odo/env/env.yaml", "Project: "+commonVar.Project)
 			})
 
+			AfterEach(func() {
+				devSession.Kill()
```

Comment:
session.Kill() will not cleanup the resources, will it? We'll need to run session.Stop() for this, no?

Reply:
We don't particularly need to cleanup resources at this point (this will be done when deleting the namespace), but to be consistent with the other tests, let's use `Stop()`

Reply:
Reverting to `Kill` as `Stop` would need a `WaitEnd()` but it is not necessary, and adding some comment on the reason why killing the process is necessary